controllers/manila_controller.go

```diff
@@ -138,11 +138,18 @@ func (r *ManilaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res
 
 	// Always patch the instance status when exiting this function so we can persist any changes.
 	defer func() {
-		// update the overall status condition if service is ready
-		if instance.IsReady() {
-			instance.Status.Conditions.MarkTrue(condition.ReadyCondition, condition.ReadyMessage)
+		// update the Ready condition based on the sub conditions
+		if instance.Status.Conditions.AllSubConditionIsTrue() {
+			instance.Status.Conditions.MarkTrue(
+				condition.ReadyCondition, condition.ReadyMessage)
+		} else {
+			// something is not ready so reset the Ready condition
+			instance.Status.Conditions.MarkUnknown(
+				condition.ReadyCondition, condition.InitReason, condition.ReadyInitMessage)
+			// and recalculate it based on the state of the rest of the conditions
+			instance.Status.Conditions.Set(
+				instance.Status.Conditions.Mirror(condition.ReadyCondition))
```

Comment:
I think `manilaapi_controller.go` needs this sort of update as well.  Looks like the `ManilaAPI` in the CI deployment never reaches the `Ready` state without it, even though all the sub-conditions are themselves `Ready`: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_manila-operator/73/pull-ci-openstack-k8s-operators-manila-operator-main-manila-operator-build-deploy-tempest/1663842943112318976/artifacts/manila-operator-build-deploy-tempest/openstack-k8s-operators-gather/artifacts/api-resources/manilaapis.yaml

Reply:
It's my bad that I thought I updated this but actually not. You are correct and this PR was incomplete and caused circular dependencies (IsReady is pending on ReadyCondition which is pending on IsReady)
I've updated the PR to cover ManilaAPI controller. Thanks for pointing that so quickly !